cmd/gpud/join/command.go

```diff
@@ -89,12 +89,15 @@ func Command(cliContext *cli.Context) (retErr error) {
 	}
 
 	// assume if not empty, it should have been persisted by the "gpud login" command
-	log.Logger.Debugw("reading private IP from state file")
-	privateIP, err := pkgmetadata.ReadMetadata(rootCtx, dbRO, pkgmetadata.MetadataKeyPrivateIP)
-	if err != nil {
-		return fmt.Errorf("failed to read private IP: %w", err)
+	privateIP := cliContext.String("private-ip")
+	if privateIP == "" {
```

Comment:
https://github.com/leptonai/gpud/blob/3ed069ff97520636248208da4f3ec10cb059567b/cmd/gpud/command/command.go#L511-L546
we provide these flags, are there similar issues?